### DISCUSSION TITLE
VS Code debug diff library

### DISCUSSION THREAD

**Discussion by Justin-E-Taylor** [codefori/vscode-ibmi#1311] [Q&A]
On my system, every source library has a corresponding object library.
There are no objects in the source libraries. VS Code debugging seems to
expect the objects will exist in the source library. Anyone know a way
around this?

TIA

**Comment by worksofliam** [comment]
Interesting idea. I am sharing this with the debug service/server owner.

For context, this is the launch configuration we're using to debug the program: https://github.com/halcyon-tech/vscode-ibmi/blob/master/src/api/debug/index.ts#L400

**Comment by mkwan01** [comment]
The source location is retrieved from the compiler generated debug information. If you have RDi, please debug the same program using RDi and see whether the source can be located without specifying additional source lookup path. If you need to specify a source lookup path for RDi, this feature is not supported yet in vscode debugging. But we do have a plan to add source lookup path feature in the future.

**Comment by Justin-E-Taylor** [comment]
In RDi, you specify the object to debug, with the library. 

Sent from AT&T Yahoo Mail on Android 
 
  On Fri, May 26, 2023 at 2:41 PM, Morris ***@***.***> wrote:   


The source location is retrieved from the compiler generated debug information. If you have RDi, please debug the same program using RDi and see whether the source can be located without specifying additional source lookup path. If you need to specify a source lookup path for RDi, this feature is not supported yet in vscode debugging. But we do have a plan to add source lookup path feature in the future.
